File: examples/async/bingx_ws_public.py
"""Subscribe to a BingX public WebSocket stream."""

import asyncio

from dcex.ws import bingx


async def main() -> None:
    """Subscribe to BingX spot trades and print one message."""
    async with bingx.public() as ws:
        await ws.subscribe_trades("BTC-USDT-SPOT")
        print(await ws.recv())


if __name__ == "__main__":
    asyncio.run(main())

File: crates/dcex/examples/bingx_ws_public.rs
use std::time::Duration;

use dcex::ws::bingx::BingxPublicWebSocket;

#[tokio::main]
async fn main() -> dcex::Result<()> {
    let mut ws = BingxPublicWebSocket::new(Duration::from_secs(10))?;
    ws.connect().await?;
    ws.subscribe_trades("BTC-USDT-SPOT").await?;
    println!("{}", ws.recv().await?);
    ws.close().await?;
    Ok(())
}
